### PR TITLE
Add configuration example with `PHP` format

### LINKED ISSUE #806: Add configuration example with `PHP` format
In the [configuration doc](https://github.com/doctrine/DoctrineMongoDBBundle/blob/4.7.x/Resources/doc/config.rst#sample-configuration), the example should be ported to PHP in addition to the current Yaml and Xml examples.

Similar to this:
https://symfony.com/doc/6.4/reference/configuration/framework.html#handler-id

### PR DESCRIPTION
**Pull request by GromNaN**
Fix #806

### CODE REVIEW

File: docs/config.rst [hunk 1]
```diff
@@ -214,6 +258,34 @@ The following configuration shows a bunch of mapping examples:
...
+        use Symfony\Config\DoctrineMongodbConfig;
+
+        return static function (DoctrineMongodbConfig $config): void {
+            $config->documentManager('default')
+                ->mapping('MyBundle1')
+                ->mapping('MyBundle2')
+                    ->type('xml')
+                ->mapping('MyBundle3')
+                    ->type('attribute')
+                    ->dir(__DIR__.'/../src/Document')
```
Comment (malarzm) [inline docs/config.rst]: I think the directories should be consistent with previous sections, so this should read `Documents/`

File: docs/config.rst [hunk 2]
```diff
@@ -214,6 +258,34 @@ The following configuration shows a bunch of mapping examples:
...
+                ->mapping('MyBundle4')
+                    ->type('xml')
+                    ->dir(__DIR__.'/../config/doctrine/mapping')
```
Comment (malarzm) [inline docs/config.rst]: This is missing Resources. Or if `Resources` is no longer encouraged we should update previous examples

File: docs/config.rst [hunk 3]
```diff
@@ -214,6 +258,34 @@ The following configuration shows a bunch of mapping examples:
...
+                ->mapping('MyBundle5')
+                    ->type('xml')
+                    ->dir('my-bundle-mappings-dir')
+                    ->alias('BundleAlias')
```
Comment (malarzm) [inline docs/config.rst]: Haven't we removed aliases and entire `Smh:Document` notation along with doctrine/common 3.0 requirement? Or is this something else?

File: docs/config.rst [hunk 4]
```diff
@@ -214,6 +258,34 @@ The following configuration shows a bunch of mapping examples:
...
+                ->mapping('MyBundle6')
+                    ->type('xml')
+                    ->dir('%kernel.project_dir%/src/vendor/DoctrineExtensions/lib/DoctrineExtensions/Documents')
```
Comment (malarzm) [inline docs/config.rst]: I see that Symfony provides a `param` function that can be used instead `%notation%`. Shall we endorse the new way or is there no community consensus yet?

File: docs/config.rst [hunk 5]
```diff
@@ -298,6 +381,24 @@ Filters may be registered with a document manager by using the following syntax:
...
         </container>
 
+    .. code-block:: php
+
+        use Symfony\Config\DoctrineMongodbConfig;
+
+        return static function (DoctrineMongodbConfig $config): void {
+            $config->documentManager('default')
+                ->filter('basic_filter')
+                    ->class('Vendor\Filter\BasicFilter')
```
Comment (malarzm) [inline docs/config.rst]: We should be encouraging `::class` notation in both `->class` calls

File: docs/config.rst [hunk 6]
```diff
@@ -684,6 +863,69 @@ Full Default Configuration
...
+            $config->autoGenerateHydratorClasses(0);
+            $config->autoGenerateProxyClasses(0);
+            $config->defaultConnection('');
+            $config->defaultDatabase('default');
+            $config->defaultDocumentManager('');
+            $config->hydratorDir('%kernel.cache_dir%/doctrine/odm/mongodb/Hydrators');
+            $config->hydratorNamespace('Hydrators');
+            $config->proxyDir('%kernel.cache_dir%/doctrine/odm/mongodb/Proxies');
+            $config->proxyNamespace('Proxies');
+            $config->fixtureLoader('Symfony\Bridge\Doctrine\DataFixtures\ContainerAwareLoader');
```
Comment (malarzm) [inline docs/config.rst]: `::class` please, everywhere

**Review by malarzm — COMMENTED** [review]
Sorry for the number of comments 🙈

File: docs/config.rst (hunk 1 above)
Comment (GromNaN) [reply docs/config.rst]: Using the constants is one of the advantages of using the PHP configuration, I think it's better to show the full feature as it should be used in the documentation.

File: docs/config.rst (hunk 4 above)
Comment (GromNaN) [reply docs/config.rst]: Yes, I'll use it.

File: docs/config.rst [hunk 7]
```diff
@@ -684,6 +863,69 @@ Full Default Configuration
...
+                ->defaultDocumentRepositoryClass('')
+                ->defaultGridfsRepositoryClass('')
+                ->repositoryFactory('')
+                ->logging(true)
+                ->autoMapping(false)
+                ->metadataCacheDriver()
+                    ->type('')
+                    ->class('')
+                    ->host('')
+                    ->port('')
```
Comment (GromNaN) [inline docs/config.rst]: This list of "defaults" is incorrect. There is no default values as this are not set by default.

**Review by franmomu — COMMENTED** [review]
I remembered that I starting changing references to bundle to fix https://github.com/doctrine/DoctrineMongoDBBundle/issues/536, we should remove them here as well (I've just created https://github.com/doctrine/DoctrineMongoDBBundle/pull/835)

**Comment by GromNaN** [comment]
@franmomu @malarzm The PR have been updated after your reviews.
I updated the first example to take the default configuration provided in the [Symfony recipe.](https://github.com/symfony/recipes-contrib/blob/main/doctrine/mongodb-odm-bundle/4.4/config/packages/doctrine_mongodb.yaml). ~The `is_bundle` part is very important since Bundle structure is no longer the default.~

File: docs/config.rst [hunk 8]
```diff
@@ -18,11 +18,11 @@ Sample Configuration
             document_managers:
                 default:
                     mappings:
-                        AcmeDemoBundle: ~
-                    filters:
-                        filter-name:
-                            class: Class\Example\Filter\ODM\ExampleFilter
-                            enabled: true
+                        App:
+                            is_bundle: false
```
Comment (GromNaN) [inline docs/config.rst:22]: The `is_bundle` and `alias` configs are not necessary for a default setup. I update the recipe: https://github.com/symfony/recipes-contrib/pull/1582

File: src/DependencyInjection/DoctrineMongoDBExtension.php [hunk 9]
```diff
@@ -462,14 +462,14 @@ private function normalizeDriverOptions(array $connection): array
      *
      *  doctrine_mongodb:
      *     mappings:
-     *         MyBundle1: ~
-     *         MyBundle2: xml
-     *         MyBundle3: { type: attribute }
-     *         MyBundle4: { type: xml, dir: Resources/config/doctrine/mapping }
-     *         MyBundle5:
+     *         App1: ~
```
Comment (GromNaN) [inline src/DependencyInjection/DoctrineMongoDBExtension.php:465]: This configs need to be updated, as `dir` is required when outside of a bundle: https://github.com/symfony/symfony/blob/7bbc9be7ea7d607fa901edb2068c2cea13d24978/src/Symfony/Bridge/Doctrine/DependencyInjection/AbstractDoctrineExtension.php#L68

Comment (franmomu) [reply src/DependencyInjection/DoctrineMongoDBExtension.php:465]: I was wondering if we should keep these examples here if we have them in the documentation.

Maybe we can remove them here and update the ones in the documentation adding the `dir` option?

File: docs/config.rst [hunk 10]
```diff
@@ -138,9 +182,9 @@ can control. The following configuration options exist for a mapping:
...
   share. This prefix should never conflict with prefixes of other defined
   mappings otherwise some of your documents cannot be found by Doctrine. This
-  option defaults to the bundle namespace + ``Document``, for example for an
-  application bundle called ``AcmeHelloBundle``, the prefix would be
-  ``Acme\HelloBundle\Document``.
+  option defaults to the application namespace + ``Document``, for example
+  for an application called ``App``, the prefix would be
+  ``App\Document``.
 
 - ``alias`` Doctrine offers a way to alias document namespaces to simpler,
```
Comment (malarzm) [inline docs/config.rst:189]: I still think we've missed deprecating this one :)

**Review by malarzm — APPROVED** [review]
Thanks for the changes! 🍻

File: src/DependencyInjection/DoctrineMongoDBExtension.php (hunk 9 above)
Comment (GromNaN) [reply src/DependencyInjection/DoctrineMongoDBExtension.php:465]: Having an example for attribute and xml, and for a bundle would be enough.

File: docs/config.rst (hunk 10 above)
Comment (GromNaN) [reply docs/config.rst:189]: Let's deprecate it for 5.1 and remove it from the doc.

**Comment by GromNaN** [comment]
Status: I need to check the examples. I'm not sure everything is correct and optimal.

**Review by IonBazan — APPROVED** [review]

**Pull request merged**